### PR TITLE
Add source files

### PR DESCRIPTION
**Pull request by ghost**
Currently available:
- Connection with encryption: none/WEP/WPA2
- UDP client/server
- TCP client

TCP server under development.

### CODE REVIEW

File: examples/ConnectNoEncryption/ConnectNoEncryption.ino [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+void setup() {
+  //Initialize serial and wait for port to open:
+  Serial.begin(9600);
+  while (!Serial) {
+    ; // wait for serial port to connect. Needed for native USB port only
+  }
+
+  // Initialize the device :
+  if (WiFi.status() == WL_NO_SHIELD) {
```
Comment (fpistm) [inline examples/ConnectNoEncryption/ConnectNoEncryption.ino:43]: Why no WiFi.begin() ?

File: examples/ConnectNoEncryption/ConnectNoEncryption.ino [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+ This example connects to an unencrypted Wifi network.
+ Then it prints the  MAC address of the Wifi shield,
+ the IP address obtained, and other network details.
+ It also set the MAC address of the device.
+ */
+#include <SPI.h>
+#include <WiFiST.h>
+
+/* Configure SPI3:
```
Comment (fpistm) [inline examples/ConnectNoEncryption/ConnectNoEncryption.ino]: Dedicated to Disco L4 IOT. Add it in the sketch name or at least in the header comment.
Same for all examples

File: examples/ConnectNoEncryption/ConnectNoEncryption.ino [hunk 3]
```diff
@@ -0,0 +1,144 @@
+/*
+
+ This example connects to an unencrypted Wifi network.
+ Then it prints the  MAC address of the Wifi shield,
```
Comment (fpistm) [inline examples/ConnectNoEncryption/ConnectNoEncryption.ino]: Not a shield in this case

File: src/ISM43362_M3G_L44_driver.cpp [hunk 4]
```diff
@@ -0,0 +1,2018 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: 2017?

File: src/ISM43362_M3G_L44_driver.cpp [hunk 5]
```diff
@@ -0,0 +1,2018 @@
...
+/**
+* @brief  Constructor.
+* @param  UARTx   : UART interface
+* @param  reset   : reset pin
+* @param  wakeup  : wakeup pin
+* @retval None
+*/
+IsmDrvClass::IsmDrvClass(HardwareSerial *UARTx,  uint8_t reset, uint8_t wakeup) {
+  /* Call UART constructor                                                   */
+  // not developped, we can't test this functionnality
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: Please explain why in the comment

File: src/ISM43362_M3G_L44_driver.cpp [hunk 6]
```diff
@@ -0,0 +1,2018 @@
...
+* @brief  Constructor.
+* @param  tx      : tx pin for USB
+* @param  rx      : rx pin for USB
+* @param  reset   : reset pin
+* @param  wakeup  : wakeup pin
+* @retval None
+*/
+IsmDrvClass::IsmDrvClass(uint8_t tx, uint8_t rx, uint8_t reset, uint8_t wakeup) {
+  /* Call USB constructor                                                    */
+  // not developped, we can't test this functionnality
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: ditto

File: src/ISM43362_M3G_L44_driver.cpp [hunk 7]
```diff
@@ -0,0 +1,2018 @@
...
+  * @param  ptr: pointer to string
+  * @param  cnt: pointer to the number of parsed digit
+  * @retval Hex value.
+  */
+uint32_t IsmDrvClass::ParseHexNumber(char* ptr, uint8_t* cnt)
+{
+    uint32_t sum = 0;
+    uint8_t i = 0;
+
+    while (CHARISHEXNUM(*ptr)) {    /* Parse number */
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: test ptr is null?

File: src/ISM43362_M3G_L44_driver.cpp [hunk 8]
```diff
@@ -0,0 +1,2018 @@
...
+  * @param  ptr: pointer to string
+  * @param  cnt: pointer to the number of parsed digit
+  * @retval integer value.
+  */
+int32_t IsmDrvClass::ParseNumber(char* ptr, uint8_t* cnt)
+{
+    uint8_t minus = 0, i = 0;
+    int32_t sum = 0;
+
+    if (*ptr == '-') {                    /* Check for minus character */
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: ditto

File: src/ISM43362_M3G_L44_driver.cpp [hunk 9]
```diff
@@ -0,0 +1,2018 @@
...
+  * @brief  Parses and returns MAC address.
+  * @param  ptr: pointer to string
+  * @param  arr: pointer to MAC array
+  * @retval None.
+  */
+void IsmDrvClass::ParseMAC(char* ptr, uint8_t* arr)
+{
+  uint8_t hexnum = 0, hexcnt;
+
+  while(* ptr) {
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: ditto

File: src/ISM43362_M3G_L44_driver.cpp [hunk 10]
```diff
@@ -0,0 +1,2018 @@
...
+  * @brief  Parses and returns IP address.
+  * @param  ptr: pointer to string
+  * @param  arr: pointer to IP array
+  * @retval None.
+  */
+void IsmDrvClass::ParseIP(char* ptr, uint8_t* arr)
+{
+  uint8_t hexnum = 0, hexcnt;
+
+  while(* ptr) {
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: ditto

File: src/ISM43362_M3G_L44_driver.cpp [hunk 11]
```diff
@@ -0,0 +1,2018 @@
...
+}
+
+/**
+  * @brief  Parses Security type.
+  * @param  ptr: pointer to string
+  * @retval Encryption type.
+  */
+ES_WIFI_SecurityType_t IsmDrvClass::ParseSecurity(char* ptr)
+{
+  if(strstr(ptr,"Open")) return ES_WIFI_SEC_OPEN;
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: ditto

File: src/ISM43362_M3G_L44_driver.cpp [hunk 12]
```diff
@@ -0,0 +1,2018 @@
...
+  * @brief  Parses ES module informations and save them in the handle.
+  * @param  ptr: pointer to string
+  * @retval None.
+  */
+void IsmDrvClass::AT_ParseInfo(uint8_t *pdata)
+{
+  char *ptr;
+  uint8_t num = 0;
+
+  ptr = strtok((char *)pdata + 2, ",");
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: test pdata is null

File: src/ISM43362_M3G_L44_driver.cpp [hunk 13]
```diff
@@ -0,0 +1,2018 @@
...
+  * @param  ptr: pointer to string
+  * @retval None.
+  */
+void IsmDrvClass::AT_ParseAP(char *pdata, ES_WIFI_APs_t *APs)
+{
+  uint8_t num = 0;
+  char *ptr;
+  APs->nbr = 0;
+
+  ptr = strtok(pdata + 2, ",");
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: ditto

File: src/ISM43362_M3G_L44_driver.cpp [hunk 14]
```diff
@@ -0,0 +1,2018 @@
...
+  * @param  APs: System Config structure
+  * @param  ptr: pointer to string
+  * @retval None.
+  */
+void IsmDrvClass::AT_ParseSystemConfig(char *pdata, ES_WIFI_SystemConfig_t *pConfig)
+{
+  uint8_t num = 0;
+  char *ptr;
+
+  ptr = strtok(pdata + 2, ",");
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: ditto

File: src/ISM43362_M3G_L44_driver.cpp [hunk 15]
```diff
@@ -0,0 +1,2018 @@
...
+  */
+void IsmDrvClass::AT_ParseAP(char *pdata, ES_WIFI_APs_t *APs)
+{
+  uint8_t num = 0;
+  char *ptr;
+  APs->nbr = 0;
+
+  ptr = strtok(pdata + 2, ",");
+
+  while ((ptr != NULL) && (APs->nbr < ES_WIFI_MAX_DETECTED_AP)) {
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: test APs is null

File: src/ISM43362_M3G_L44_driver.cpp [hunk 16]
```diff
@@ -0,0 +1,2018 @@
...
+{
+  uint8_t num = 0;
+  char *ptr;
+
+  ptr = strtok(pdata + 2, ",");
+
+  while (ptr != NULL) {
+    switch (num++) {
+    case 0:
+      pConfig->Configuration = ParseNumber(ptr, NULL);
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp:423]: test pconfig is null

File: src/ISM43362_M3G_L44_driver.cpp [hunk 17]
```diff
@@ -0,0 +1,2018 @@
...
+  * @param  NetSettings: settings
+  * @param  pdata: pointer to data
+  * @retval None.
+  */
+void IsmDrvClass::AT_ParseConnSettings(char *pdata, ES_WIFI_Network_t *NetSettings)
+{
+  uint8_t num = 0;
+  char *ptr;
+
+  ptr = strtok(pdata + 2, ",");
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: pdata is null?

File: src/ISM43362_M3G_L44_driver.cpp [hunk 18]
```diff
@@ -0,0 +1,2018 @@
...
+
+  while (ptr != NULL) {
+    switch (num++) {
+    case 0:
+      strncpy((char *)NetSettings->SSID,  ptr, ES_WIFI_MAX_SSID_NAME_SIZE + 1);
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp:494]: NetSettings is null?

File: src/ISM43362_M3G_L44_driver.cpp [hunk 19]
```diff
@@ -0,0 +1,2018 @@
...
+  * @param  pdata: pointer to data.
+  * @param  ConnSettings: struct settings.
+  * @retval None.
+  */
+void IsmDrvClass::AT_ParseTrSettings(char *pdata, ES_WIFI_Conn_t *ConnSettings)
+{
+  uint8_t num = 0;
+  char *ptr;
+
+  ptr = strtok(pdata + 2, ",");
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: pdata is null?

File: src/ISM43362_M3G_L44_driver.cpp [hunk 20]
```diff
@@ -0,0 +1,2018 @@
...
+      // Protocol identifier
+      ptr++;
+      break;
+
+    case 1:
+      ParseIP(ptr, ConnSettings->RemoteIP);
+      break;
+
+    case 2:
+      ConnSettings->LocalPort = ParseNumber(ptr, NULL);
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp:581]: ConnSettings is null?

File: src/ISM43362_M3G_L44_driver.cpp [hunk 21]
```diff
@@ -0,0 +1,2018 @@
...
+}
+
+/**
+  * @brief  Execute AT command.
+  * @param  cmd: pointer to command string
+  * @param  pdata: pointer to returned data
+  * @retval Operation Status.
+  */
+ES_WIFI_Status_t IsmDrvClass::AT_ExecuteCommand(uint8_t* cmd, uint8_t *pdata)
+{
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp:611]: param is null?

File: src/ISM43362_M3G_L44_driver.cpp [hunk 22]
```diff
@@ -0,0 +1,2018 @@
...
+/**
+  * @brief  Execute AT command with data.
+  * @param  cmd: pointer to command string
+  * @param  pcmd_data: pointer to binary data
+  * @param  len: binary data length
+  * @param  pdata: pointer to returned data
+  * @retval Operation Status.
+  */
+ES_WIFI_Status_t IsmDrvClass::AT_RequestSendData(uint8_t* cmd,
+uint8_t *pcmd_data, uint16_t len, uint8_t *pdata)
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp]: ditto

File: src/ISM43362_M3G_L44_driver.cpp [hunk 23]
```diff
@@ -0,0 +1,2018 @@
...
+
+/**
+  * @brief  Parses Received long data length.
+  * @param  pdata: payload
+  * @param  Reqlen : requested Data length.
+  * @param  ReadData : pointer to received data length.
+  * @retval Operation Status.
+  */
+ES_WIFI_Status_t IsmDrvClass::ReceiveLongDataLen(char *pdata, uint16_t Reqlen,
+ uint16_t *ReadData)
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.cpp:733]: ditto

File: src/ISM43362_M3G_L44_driver.h [hunk 24]
```diff
@@ -0,0 +1,446 @@
...
+
+typedef struct
+{
+  uint32_t Port;
+  uint32_t BaudRate;
+  uint32_t DataWidth;
+  uint32_t Parity;
+  uint32_t StopBits;
+  uint32_t Mode;
+}ES_WIFI_UARTConfig_t;
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.h:225]: Not used in the code?

File: src/ISM43362_M3G_L44_driver.h [hunk 25]
```diff
@@ -0,0 +1,446 @@
...
+
+/* Access point configuration */
+typedef struct {
+  uint8_t SSID[ES_WIFI_MAX_SSID_NAME_SIZE + 1];  /*!< Network public name for ESP AP mode */
+  uint8_t Pass[ES_WIFI_MAX_PSWD_NAME_SIZE + 1];  /*!< Network password for ESP AP mode */
+  ES_WIFI_SecurityType_t Security;          /*!< Security of Wi-Fi spot. This parameter can be a value of \ref ESP8266_Ecn_t enumeration */
+  uint8_t Channel;                          /*!< Channel Wi-Fi is operating at */
+  uint8_t MaxConnections;                   /*!< Max number of stations that are allowed to connect to ESP AP, between 1 and 4 */
+  uint8_t Hidden;                           /*!< Set to 1 if network is hidden (not broadcast) or zero if noz */
+} ES_WIFI_APConfig_t;
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.h]: Not used in the code?

File: src/ISM43362_M3G_L44_driver.h [hunk 26]
```diff
@@ -0,0 +1,446 @@
...
+typedef struct {
+  ES_WIFI_ConnType_t Type;
+  uint8_t            Number;
+  uint16_t           RemotePort;
+  uint8_t            RemoteIP[4];
+  uint8_t            *PublishTopic;
+  uint8_t            *SubscribeTopic;
+  uint8_t            *ClientID;
+  uint8_t            MQTTMode;
+} ES_WIFI_AWS_Conn_t;
```
Comment (fpistm) [inline src/ISM43362_M3G_L44_driver.h]: Not used in the code?

File: src/WiFiClientST.cpp [hunk 27]
```diff
@@ -0,0 +1,284 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/WiFiClientST.cpp]: 2017?

File: src/WiFiClientST.h [hunk 28]
```diff
@@ -0,0 +1,93 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/WiFiClientST.h]: 2017?

File: src/WiFiST.cpp [hunk 29]
```diff
@@ -0,0 +1,351 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/WiFiST.cpp]: 2017?

File: src/WiFiST.h [hunk 30]
```diff
@@ -0,0 +1,105 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/WiFiST.h]: 2017?

File: src/WiFiUdpST.cpp [hunk 31]
```diff
@@ -0,0 +1,295 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/WiFiUdpST.cpp]: 2017?

File: src/WiFiUdpST.cpp [hunk 32]
```diff
@@ -0,0 +1,295 @@
...
+}
+
+/**
+ * @brief  Finish with the UDP Socket
+ * @param  None
+ * @retval None
+ */
+void WiFiUDP::stop()
+{
+  if (_sock == NO_SOCKET_AVAIL)
```
Comment (fpistm) [inline src/WiFiUdpST.cpp]: (_sock != NO_SOCKET_AVAIL) to avoid a return

File: src/WiFiUdpST.cpp [hunk 33]
```diff
@@ -0,0 +1,295 @@
...
+ * @param  buffer: data to write
+ * @param  size: size of data to write
+ * @retval size of write data
+ */
+size_t WiFiUDP::write(const uint8_t *buffer, size_t size)
+{
+  uint16_t SentLen = 0;   // Number of data really send
+  uint8_t *temp = (uint8_t*)buffer;
+
+   DrvWiFi->ES_WIFI_SendResp(_sock, temp, size, &SentLen, WIFI_TIMEOUT);
```
Comment (fpistm) [inline src/WiFiUdpST.cpp]: one extra space

File: src/WiFiUdpST.h [hunk 34]
```diff
@@ -0,0 +1,92 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/WiFiUdpST.h]: 2017?

File: src/driver.h [hunk 35]
```diff
@@ -0,0 +1,81 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/driver.h]: 2017?

File: src/es_wifi_conf.h [hunk 36]
```diff
@@ -0,0 +1,88 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/es_wifi_conf.h]: 2017?

File: src/spi_drv.cpp [hunk 37]
```diff
@@ -0,0 +1,341 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/spi_drv.cpp]: 2017?

File: src/spi_drv.cpp [hunk 38]
```diff
@@ -0,0 +1,341 @@
...
+
+  pinMode(spiIRQPin, INPUT);
+
+  pinMode(resetPin, OUTPUT);
+
+  pinMode(csPin, OUTPUT);
+  digitalWrite(csPin, HIGH);
+
+  /* 80/8= 10MHz (Inventek WIFI module supportes up to 20MHz)                */
+  Settings_43362 = new SPISettings(SPI_CLOCK_DIV8, MSBFIRST, SPI_MODE0);
```
Comment (fpistm) [inline src/spi_drv.cpp]: Where the 80/8 comes from?
is it the SystemCoreClock  of the boards?
What happen if I change the clock config?

File: src/spi_drv.h [hunk 39]
```diff
@@ -0,0 +1,93 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/spi_drv.h]: 2017?

File: src/wifi_drv.h [hunk 40]
```diff
@@ -0,0 +1,179 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/wifi_drv.h]: 2017?

File: src/wifi_drv.h [hunk 41]
```diff
@@ -0,0 +1,179 @@
...
+  ES_WIFI_STATUS_REQ_DATA_STAGE = 1,
+  ES_WIFI_STATUS_ERROR          = 2,
+  ES_WIFI_STATUS_TIMEOUT        = 3,
+  ES_WIFI_STATUS_IO_ERROR       = 4,
+}ES_WIFI_Status_t;
+
+typedef enum {
+  ES_WIFI_MODE_SINGLE           = 0,
+  ES_WIFI_MODE_MULTI            = 1,
+}ES_WIFI_ConnMode_t;
```
Comment (fpistm) [inline src/wifi_drv.h:92]: Not used in the code?

File: src/wl_definitions.h [hunk 42]
```diff
@@ -0,0 +1,83 @@
...
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'
+     "-'    "--
+
+----------------------------------------------------------------------------
+
+Copyright (c) <2014>, <Wi6labs>
```
Comment (fpistm) [inline src/wl_definitions.h]: 2017?

**Review by fpistm — CHANGES_REQUESTED** [review]
See comment.
Moreover not functional with Arduino_Core_STM32  since https://github.com/stm32duino/Arduino_Core_STM32/commit/18cce4b6f6bd32d379726b523694de1d2a189a18

File: examples/ConnectNoEncryption/ConnectNoEncryption.ino (hunk 1 above)
Comment (ghost) [reply examples/ConnectNoEncryption/ConnectNoEncryption.ino:43]: We kept the Arduino WiFi API mechanism.
`Begin()` is call after to enable the connection.

File: src/ISM43362_M3G_L44_driver.h (hunk 24 above)
Comment (ghost) [reply src/ISM43362_M3G_L44_driver.h:225]: But could be used in case of the UART driver.

File: src/ISM43362_M3G_L44_driver.h (hunk 25 above)
Comment (ghost) [reply src/ISM43362_M3G_L44_driver.h]: AP config not implemented.

File: src/ISM43362_M3G_L44_driver.h (hunk 26 above)
Comment (ghost) [reply src/ISM43362_M3G_L44_driver.h]: AWS mode not implemented in fact.

File: src/wifi_drv.h (hunk 41 above)
Comment (ghost) [reply src/wifi_drv.h:92]: Because server is not implemented yet.

File: src/spi_drv.cpp (hunk 38 above)
Comment (ghost) [reply src/spi_drv.cpp]: Comes from the ST library. SPI_CLOCK_DIV8 will be replace by the frequency so nothing will happen if you change the clock config.

File: examples/ConnectNoEncryption/ConnectNoEncryption.ino (hunk 1 above)
Comment (fpistm) [reply examples/ConnectNoEncryption/ConnectNoEncryption.ino:43]: ok

**Comment by ghost** [comment]
Server API added to the library.
Fully tested on rev D.

**Comment by ghost** [comment]
Add a better status management. 
Fix reset issue when call status after begin.

**Review by fpistm — APPROVED** [review]

**Pull request merged**